pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -420,10 +422,33 @@ def get_treeinfo_data(remote, remote_url):
 
         return treeinfo_serialized
 
+    def get_sync_details(remote, url, sync_policy, version):
+        with tempfile.TemporaryDirectory("."):
+            try:
+                result = get_repomd_file(remote, url)
+                repomd_path = result.path
+                repomd = cr.Repomd(repomd_path)
+                repomd_checksum = get_sha256(repomd_path)
+            except Exception:
+                repomd_checksum = None
```

Comment:
Why do we continue if we can't properly read or get repomd.xml?

Reply:
The comparison of the sync parameters is happening externally, the only way to not continue would be to raise an exception.  Which feels a little messy, at least for now.  The net effect is still that the optimization decision will fail.

Also it just feels like a weird place to see an error message coming from, since it's a non-critical step.

Reply:
Unless there is an unlikely situation when sync somehow succeeded with the checksum being `None` .or we have a bug in our code and this code in try/except always fails; so for every sync None is saved and compared.

I agree with you about a weird place to fail and that it feels messy though. Not a blocker but it;s confusing as a reader what the intention is here.